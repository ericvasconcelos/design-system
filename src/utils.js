export { arrayCombinations } from 'utils/arrayCombinations'
export { getBackgroundColor } from 'utils/getBackgroundColor'
export { getBorder } from 'utils/getBorder'
export { getButtonStyle } from 'utils/getButtonStyle'
export { getBorderRadius } from 'utils/getBorderRadius'
export { getColorFromName } from 'utils/getColorFromName'
export { getGuttersNames } from 'utils/getGuttersNames'
export { getIconNames } from 'utils/getIconNames'
export { getPadding } from 'utils/getPadding'
export { getPaddingNames } from 'utils/getPaddingNames'
export { getTypography } from 'utils/getTypography'
export { parsePalleteName } from 'utils/parsePalleteName'
export { styleColorValidator } from 'utils/styleColorValidator'
export { useAriaProps } from 'utils/useAriaProps'

export { DateUtils } from 'utils/DateUtils'
export { QueryUtils } from 'utils/QueryUtils'
export { ObjectUtils }  from 'utils/ObjectUtils'
export { CookiesUtils } from 'utils/CookiesUtils'
export { MoneyUtils } from 'utils/MoneyUtils'
export { NumberUtils } from 'utils/NumberUtils'