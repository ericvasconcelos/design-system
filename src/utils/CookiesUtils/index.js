export class CookiesUtils {
  static toObject(string = '') {
    const cookies = {}
    if (string !== '') {
      const parts = string.split('; ')
      parts.forEach(part => {
        const [key, value] = part.split('=')
        cookies[key] = value
      })
    }
    return cookies
  }
}
