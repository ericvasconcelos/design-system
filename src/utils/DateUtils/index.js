export class DateUtils {
  static toDate(string, format = 'pt-br') {
    let day = ''
    let month = ''
    let year = ''

    switch (format) {
      case 'pt-br':
        [day, month, year] = string?.split('/')
        break
      case 'iso':
        [year, month, day] = string?.split('-')
        break
      default:
        throw new Error('Format invalid')
    }

    return new Date(year, parseInt(month, 10) - 1, day)
  }

  static getAge(date) {
    if (!(date instanceof Date)) {
      throw new TypeError('Date invalid')
    }

    const difference = Date.now() - date.getTime()

    return Math.abs(new Date(difference).getUTCFullYear() - 1970)
  }

  static toString(date, format = 'pt-br') {
    if (!(date instanceof Date)) {
      throw new TypeError('Date invalid')
    }

    const day = date.getDate() < 10 ? `0${date.getDate()}` : date.getDate()
    const month = date.getMonth() + 1 < 10 ? `0${date.getMonth() + 1}` : date.getMonth() + 1
    const year = date.getFullYear()

    switch (format) {
      case 'pt-br':
        return `${day}/${month}/${year}`
      case 'iso':
        return `${year}-${month}-${day}`
      default:
        throw new Error('Format invalid')
    }
  }

  static validate(date, format = 'pt-br') {
    const monthDaysMap = {
      '01': 31,
      '02': 29,
      '03': 31,
      '04': 30,
      '05': 31,
      '06': 30,
      '07': 31,
      '08': 31,
      '09': 30,
      10: 31,
      11: 30,
      12: 31
    }
    let isValid = false

    let day = ''
    let month = ''
    let year = ''

    switch (format) {
      case 'pt-br':
        [day, month, year] = date?.split('/')
        break
      case 'iso':
        [year, month, day] = date?.split('-')
        break
      default:
        throw new Error('Format invalid')
    }

    isValid = isValid || parseInt(day, 10) <= monthDaysMap[month]
    isValid = isValid && parseInt(year, 10) > 1900

    return isValid
  }

  static convertMonthToYear(month) {
    return parseInt(month, 10) / 12
  }

  static formatPeriod(period, periodType) {
    const periodTypeMap = {
      monthly: { singular: 'mÃªs', plural: 'meses' },
      yearly: { singular: 'ano', plural: 'anos' },
      daily: { singular: 'dia', plural: 'dias' }
    }
    return `${period} ${periodTypeMap[periodType][period < 2 ? 'singular' : 'plural']}`
  }
}
