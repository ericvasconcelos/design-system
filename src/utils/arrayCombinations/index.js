export const arrayCombinations = (...args) => {
  const returnable = []
  const max = args.length - 1
  function helper(arr, i) {
    for (let j = 0, { length } = args[i]; j < length; j++) {
      const newArray = arr.slice(0) // clone arr
      newArray.push(args[i][j])
      if (i === max) returnable.push(newArray)
      else helper(newArray, i + 1)
    }
  }
  helper([], 0)
  return returnable
}
