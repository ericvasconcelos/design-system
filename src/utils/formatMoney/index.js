import { toMoney, toNumber } from 'vanilla-masker'

export const formatMoney = entry => {
  const moneyFormatter = {
    precision: 2,
    separator: ',',
    delimiter: '.',
    unit: 'R$'
  }

  return toMoney(toNumber(entry), moneyFormatter)
}
