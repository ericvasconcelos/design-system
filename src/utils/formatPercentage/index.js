import { toMoney, toNumber } from 'vanilla-masker'

export const formatPercentage = entry => {
  const percentageFormatter = {
    precision: 1,
    separator: ',',
    suffixUnit: '%'
  }
  return toMoney(toNumber(entry), percentageFormatter)
}
