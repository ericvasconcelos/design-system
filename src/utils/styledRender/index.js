import React from 'react'

import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import { theme } from 'styles/theme'

export const styledRender = (children, options = {}) => {
  const rendered = render(<ThemeProvider theme={theme}>{children}</ThemeProvider>, options)
  return {
    ...rendered,
    rerender: (ui, rerenderOptions = {}) =>
      styledRender(ui, { container: rendered.container, ...rerenderOptions })
  }
}
