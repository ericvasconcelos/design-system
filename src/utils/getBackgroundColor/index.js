import { getColorFromName } from 'utils/getColorFromName'

export const getBackgroundColor = ({ theme, styleColor }) =>
  `background-color: ${getColorFromName({ theme, styleColor })};`
