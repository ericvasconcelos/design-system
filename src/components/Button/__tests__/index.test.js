import React from 'react'

import { styledRender } from 'utils/styledRender'

import { Button } from '../index.js'

describe('should render the component correctly when', () => {
  it('has basic props', () => {
    const { getByTestId } = styledRender(<Button onClick={() => {}}>Test</Button>)
    const component = getByTestId('Button')
    expect(component).toMatchSnapshot()
  })
})
