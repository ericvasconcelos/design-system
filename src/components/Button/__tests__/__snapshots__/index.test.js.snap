// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the component correctly when has basic props 1`] = `
.c0 {
  border: 2px solid #003FF2;
  border-radius: 0.5rem;
  background-color: #003FF2;
  padding: 0.5rem 1rem;
  color: #FFFFFF;
  font-size: 0.625rem;
  font-weight: 500;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
  width: 100%;
  text-align: left;
  gap: 0.5rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
  height: 44px;
  opacity: 1;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  text-transform: uppercase;
}

.c0 >div {
  margin: none;
}

.c0:hover:enabled {
  background-color: #8DC6FF;
  border: 2px solid #8DC6FF;
  color: #FFFFFF;
  font-size: 0.625rem;
  font-weight: 500;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c0:hover:enabled >div {
  background: #FFFFFF;
}

.c0:active:enabled {
  background-color: #1C49D9;
  border: 2px solid #1C49D9;
  color: #FFFFFF;
  font-size: 0.625rem;
  font-weight: 500;
  line-height: 1rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  font-family: 'DM Sans',sans-serif;
}

.c0:active:enabled >div {
  background: #FFFFFF;
}

<button
  class="c0"
  data-testid="Button"
  id=""
  tabindex="0"
  type="button"
>
  Test
</button>
`;
